### PR TITLE
feat(ecommerce): add ecommerce web app

### LINKED ISSUE #143: FRONTEND: Ecommerce web app
### Frontend App Title

Ecommerce web app

### Overview

A user facing ecommerce web application which would allow the users to browser through products, add them to cart, and order them.

### Detailed Features

Application Features: 

1. Allows user to register by email or Google SSO.
2. View Categories, View Products In a Category, View All Products, View Product Details, Search for a product.
3. Add Products to Cart, Update Product Quantity In Cart, Delete Product from Cart.
4. Add address, Update Address, Delete an address. 
5. View eligible coupon codes.
6. Apply Eligible Coupon Codes.
7. Pay via PayPal.
8. Update profile, Change Password.
9. View Orders.
10. Supports Multilingualism, along with right to left languages.

Design Inspired From:  https://www.figma.com/community/file/1219312065205187851

### Tech Stack Used

ReactJS, Redux, Tailwind CSS

### Built For Platform

web

### Project Path

$ROOT_FOLDER/examples/apps/ecommerce/web/react-vite-redux-tailwind

### PR DESCRIPTION
**Pull request by amitHarwani**
Closes #143 
Features Implemented: 
1. Register by email or Google SSO.
2. View Categories, View Products In a Category, View All Products, View Product Details, Search for a product.
3. Add Products to Cart, Update Product Quantity In Cart, Delete Product from Cart.
4. Add address, Update Address, Delete an address. 
5. View eligible coupon codes.
6. Apply Eligible Coupon Codes.
7. Pay via PayPal.
8. Update profile, Change Password.
9. View Orders.
10. Multilingualism.

### CODE REVIEW

**Comment by wajeshubham** [comment]
@amitHarwani Thank you for the PR, the application looks really good. Could you please help us with the hosted link of the project so that we get an overview of the application.

**Comment by jwala-anirudh** [comment]
After performing sign in with google action, the redirect back leads to 404

Screenshot 2024-04-26 at 9 08 09 AM

**Comment by amitHarwani** [comment]
> @amitHarwani Thank you for the PR, the application looks really good. Could you please help us with the hosted link of the project so that we get an overview of the application. Also, are we planning to add admin panel in this? I cannot see the product and category creation pages.

Hi, @wajeshubham, Thank you for the message, Yes I do plan to complete the project with the admin module, I am currently looking into some designs for the same.
Hosted Link: https://ecommerce-client-tsy6.onrender.com/, Just some points on the hosted app: 
 * Render by default stops the web service after 15 mins of inactivity in its free tier, hence the first load may have a lag.
 * Paypal integration is done with a sandbox account since I don't have a business account.
 * Mailtrap is working under the test functionality.

**Comment by amitHarwani** [comment]
> After performing sign in with google action, the redirect back leads to 404
> 
> Screenshot 2024-04-26 at 9 08 09 AM

Hi, @jwala-anirudh, Thanks for pointing this out, I forgot to mention that the CLIENT_SSO environment variable has to be changed to the <frontend_endpoint>/, I have updated the same in the README file.

File: src/controllers/apps/auth/user.controllers.js [hunk 1]
```diff
@@ -339,9 +339,7 @@ const forgotPasswordRequest = asyncHandler(async (req, res) => {
       // ! NOTE: Following link should be the link of the frontend page responsible to request password reset
       // ! Frontend will send the below token with the new password in the request body to the backend reset password endpoint
       // * Ideally take the url from the .env file which should be teh url of the frontend
-      `${req.protocol}://${req.get(
-        "host"
-      )}/api/v1/users/reset-password/${unHashedToken}`
+      `${process.env.FORGOT_PASSWORD_REDIRECT_URL}/${unHashedToken}`
```
Comment (wajeshubham) [inline src/controllers/apps/auth/user.controllers.js:342]: Please update the `.env.sample` file with newly added env var with necessary comment to understand what env var does.

**Review by wajeshubham — CHANGES_REQUESTED** [review]

File: examples/apps/ecommerce/web/react-vite-redux-tailwind/public/fonts/poppins/OFL.txt
[the remark this answers is not in the document]
Comment (amitHarwani) [reply examples/apps/ecommerce/web/react-vite-redux-tailwind/public/fonts/poppins/OFL.txt:1]: Updated to use fonts from link tag.

File: src/controllers/apps/auth/user.controllers.js (hunk 1 above)
Comment (amitHarwani) [reply src/controllers/apps/auth/user.controllers.js:341]: Updated .env.sample with the comments.

**Review by wajeshubham — APPROVED** [review]

**Comment by wajeshubham** [comment]
Thank you @amitHarwani for the contribution

**Pull request merged**